### ISSUE TITLE
Active questions are disappearing from the site. 

### ISSUE THREAD

**Issue by bronwen9** [publiclab/plots2#2510]
There are at least two questions with active threads that have been deleted from the site. 

One was from user cfastie: https://publiclab.org/profile/cfastie, who had posted a question about creating website alerts. He also pointed to this:

"This is the missing node (my question from yesterday).
https://publiclab.org/notes/cfastie/03-18-2018/how-do-i-do-website-change-detection-and-notification
Although I don't know why it is a "note" instead of a "question." 

The other was from user MarcioPamplona: https://publiclab.org/profile/MarcioPamplona, who had a post about methods for detecting moisture content in wood. They re-posted their question here: https://publiclab.org/questions/MarcioPamplona/03-19-2018/wood-moisture-measurement but the original thread appears to be lost.

**Comment by jywarren** [comment]
This is a known issue from this weekend which is not code related, thanks! Two nodes were dropped from the database to our knowledge; please paste in here any missing node urls and we're in the process of a recovery. Thanks! @icarito

**Comment by jywarren** [comment]
https://publiclab.org/notes/MarcioPamplona/03-19-2018/wood-moisture-measurement

Node IDs: 15975

Chris's was: 15972

**Comment by jywarren** [comment]
```
I live in a town where malathion and permethrin are sprayed in a fog along roads to kill mosquitoes in the summer. They are required to notify the public when they spray, but they do this only by posting on a web page. In order to alert the community that spraying will happen, I would like to [monitor this page](https://blsgmosquito.wordpress.com/whats-new/) or maybe one linked there. How can I get an email every time this page is changed? Spraying happens after dark, and the decision to spray might happen late in the day. So I might want to check the page every ten minutes in the afternoon and evening.
```

**Comment by jywarren** [comment]
`How do I do website change detection and notification?`

**Comment by jywarren** [comment]
`Quantitative analysis of wood moisture`

```
I need some advice on how to design a system to measure the amount of moisture in wood. 
Measurement by reflection, directly on the surface of the material.
Do you have to calibrate for each type of material or can we directly measure the amount of water molecules in the material?

Thanks in advance
```

**Comment by jywarren** [comment]
Oh, sorry, MarcioPamplona's nid was actually `15971`

**Comment by jywarren** [comment]
made more complex bc i believe this spam post was then made, claiming the `15971` `nid` - that's why that one has all the views and comments and such. I'm going to delete that node entry since it's spam and replace it with the lost node.

**Comment by jywarren** [comment]
OK, that was complex, but I think i've fully deleted the spam post, and MarcioPamplona's post is now available at the original URL with all the original ancillary content -- tags, comments, answers, URL. Please help me out if I missed anything. I'm doing same for Chris's in a second.

**Comment by icarito** [comment]
Thank you Jeff!

**Comment by jywarren** [comment]
NP! For reference, I retrieved the titles and body text (although not Markdown, so that'd be tougher with a complex markdown post) from the email notifications I got (I get every one, as do a number of people: https://publiclab.org/tag/everything

Conveniently the titles and body text are actually saved in the Revision table, so I instead then went in and used the old NIDs to search for `Revisions.find_by(nid: ___)` and then rebuilt the node records from that. 

I found the lost NIDs by checking the moderator email "Spam" and "Approve" links, which include `nid` in the URL. This helped. 

Finally for the two cases where the nodes had been overwritten by new nodes, we got lucky that they were clearly spam nodes, so I overwrote them with the original `uid`, `path` and `title` fields, and changed their status back to 1.

The path I regenerated with `node.path = node.generate_path` and then saved. 

Not simple, but everything was recoverable!
